lib/logstash/util/compress.rb

```diff
@@ -0,0 +1,117 @@
+  module Zip
+
+    extend self
+
+    # Extract a zip file into a destination directory.
+    # @param source [String] The location of the file to extract
+    # @param target [String] Where you do want the file to be extracted
+    # Raise IOError If the target directory already exist
+    def extract(source, target)
+      raise IOError.new("Directory #{target} exist") if File.exist?(target)
+      ::Zip::File.open(source) do |zip_file|
+        zip_file.each do |file|
+          path = File.join(target, file.name)
+          FileUtils.mkdir_p(File.dirname(path))
+          zip_file.extract(file, path)
+        end
+      end
+    end
+
+    # Compress a directory into a zip file
+    # @param dir [String] The directory to be compressed
+    # @param target [String] Destination to save the generated file
+    # Raise IOError If the target file already exist
+    def compress(dir, target)
+      raise IOError.new("File #{target} exist") if File.exist?(target)
+      ::Zip::File.open(target, ::Zip::File::CREATE) do |zipfile|
+        Dir.glob("#{dir}/**/*").each do |file|
+          path_in_zip = file.gsub("#{dir}/","")
+          zipfile.add(path_in_zip, file)
+        end
+      end
+    end
+  end
+
+  module Tar
+
+    extend self
+
+    # Extract a tar.gz file into a destination directory.
+    # @param source [String] The location of the file to extract
+    # @param target [String] Where you do want the file to be extracted
+    # Raise IOError If the target directory already exist
+    def extract(file, target)
+      raise IOError.new("Directory #{target} exist") if File.exist?(target)
+
+      FileUtils.mkdir(target)
+      Zlib::GzipReader.open(file) do |gzip_file|
+        Gem::Package::TarReader.new(gzip_file) do |tar_file|
+          tar_file.each do |entry|
+            target_path = File.join(target, entry.full_name)
+
+            if entry.directory?
+              FileUtils.mkdir_p(target_path)
+            else # is a file to be extracted
+              File.open(target_path, "wb") do |f|
+                f.write(entry.read)
+              end
+            end
+          end
+        end
+      end
+    end
+
+    # Compress a directory into a tar.gz file
+    # @param dir [String] The directory to be compressed
+    # @param target [String] Destination to save the generated file
+    # Raise IOError If the target file already exist
+    def compress(dir, target)
+      raise IOError.new("File #{target} exist") if File.exist?(target)
+
+      tar_file = Tempfile.new(File.basename(target))
+
+      ::Gem::Package::TarWriter.new(tar_file) do |tar|
+        Dir.glob("#{dir}/**/*").each do |file|
+          name = file.gsub("#{dir}/","")
+          mode = File.stat(file).mode
+
+          if File.directory?(file)
+            tar.mkdir(name, mode)
+          else # is a file to be added
+            tar.add_file(name,mode) do |out|
+              File.open(file, "rb") do |f|
+                out.write(f.read)
+              end
```

Comment:
I think we should think more strongly on upgrading the style guide + using robocop so this style things are more clear to everyone, otherwise we end up with personal experience and/or taste. A +1 on upgrading the style guide :+1: